### PR TITLE
bug fix for parallel

### PR DESCRIPTION
**Pull request by oHunewald**

### CODE REVIEW

File: test/Test_SOM_parallel.jl [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+som2 = initSOM_parallel(df_som, 10, 10)
+
+@time som2 = trainSOM_parallel(som2, df_som, size(df_som)[1], epochs = 10)
+
+mywinners = mapToSOM(som2, df_som)
+CSV.write("cell_clustering_som.csv", mywinners)
+
+# myfreqs = SOM.classFrequencies(som2, daf.fcstable, :sample_id)
+
+codes = som2.codes
```
Comment (laurentheirendt) [inline test/Test_SOM_parallel.jl:23]: May you use `@test` here?

File: test/Test_SOM_parallel.jl [hunk 2]
```diff
@@ -0,0 +1,36 @@
+using Distributed
+
+p = addprocs(2)
+
+# @everywhere using Pkg
```
Comment (laurentheirendt) [inline test/Test_SOM_parallel.jl]: These lines can be removed.

File: test/Test_SOM_parallel.jl [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+df_som = vcat(df_som, df_som)
+
+som2 = initSOM_parallel(df_som, 10, 10)
+
+@time som2 = trainSOM_parallel(som2, df_som, size(df_som)[1], epochs = 10)
+
+mywinners = mapToSOM(som2, df_som)
+CSV.write("cell_clustering_som.csv", mywinners)
+
+# myfreqs = SOM.classFrequencies(som2, daf.fcstable, :sample_id)
```
Comment (laurentheirendt) [inline test/Test_SOM_parallel.jl:21]: Can this line be removed?

File: test/Test_SOM_parallel.jl [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+p = addprocs(2)
+
+# @everywhere using Pkg
+# @everywhere Pkg.add("DistributedArrays")
+# @everywhere Pkg.add("DataFrames")
+# @everywhere Pkg.add("Distances")
+
+@everywhere using DistributedArrays
+@everywhere using GigaSOM
+# @everywhere using DataFrames
```
Comment (laurentheirendt) [inline test/Test_SOM_parallel.jl]: These lines can be removed.

File: src/parallel_som.jl [hunk 5]
```diff
@@ -549,5 +516,40 @@ function convertTrainingData(data)::Array{Float64,2}
...
+    prt_size = ceil(len_matrix / nWorkers)
+
+    prt_matrix = Dict()
+    for (p, pid) in enumerate(workers())
+        prt_matrix[pid] = matrix_train[1:prt_size, : ]
+    end
+
+
+    #
+    # w = workers()
```
Comment (laurentheirendt) [inline src/parallel_som.jl]: Can these lines be removed?

File: src/parallel_som.jl [hunk 6]
```diff
@@ -549,5 +516,40 @@ function convertTrainingData(data)::Array{Float64,2}
     return train
 end
 
+function splitMatrix(matrix_train, nWorkers)
```
Comment (laurentheirendt) [inline src/parallel_som.jl]: May you add the docstring for this function?

File: src/parallel_som.jl [hunk 7]
```diff
@@ -110,47 +106,46 @@ function trainSOM_parallel(som::Som, train::Any, len;
...
+            Δr = 0.0
+        else
+            Δr = (r-1.0) / epochs
+        end
     else
-     Δr = 0.0
+        Δr = 0.0
     end
 
+    nWorkers = nprocs()
```
Comment (laurentheirendt) [inline src/parallel_som.jl:122]: What is the difference with `workers()`?

File: src/parallel_som.jl [hunk 8]
```diff
@@ -89,16 +89,12 @@ function trainSOM_parallel(som::Som, train::Any, len;
                      kernelFun::Function = gaussianKernel, r = 0.0,
                      rDecay = true, epochs = 10)
 
-    println(first(train, 6))
+    # println(first(train, 6))
```
Comment (laurentheirendt) [inline src/parallel_som.jl]: May you introduce a `printLevel`?

**Review by laurentheirendt — CHANGES_REQUESTED** [review]

File: test/Test_SOM_parallel.jl (hunk 2 above)
Comment (oHunewald) [reply test/Test_SOM_parallel.jl]: done

File: test/Test_SOM_parallel.jl (hunk 3 above)
Comment (oHunewald) [reply test/Test_SOM_parallel.jl:21]: done

File: test/Test_SOM_parallel.jl (hunk 4 above)
Comment (oHunewald) [reply test/Test_SOM_parallel.jl]: done

File: src/parallel_som.jl (hunk 5 above)
Comment (oHunewald) [reply src/parallel_som.jl]: done

File: src/parallel_som.jl (hunk 6 above)
Comment (oHunewald) [reply src/parallel_som.jl]: removed function

File: src/parallel_som.jl (hunk 7 above)
Comment (oHunewald) [reply src/parallel_som.jl:122]: ```workers()``` actually returns an array of worker id's
```nprocs()``` returns the number of workers

File: src/parallel_som.jl (hunk 8 above)
Comment (oHunewald) [reply src/parallel_som.jl]: removed

**Review by laurentheirendt — APPROVED** [review]

**Comment by laurentheirendt** [comment]
Nice! Thanks @oHunewald 👍

**Pull request merged**